liblangutil/Scanner.cpp

```diff
@@ -812,6 +882,14 @@ Token Scanner::scanString(bool const _isUnicode)
 	}
 	if (m_char != quote)
 		return setError(ScannerError::IllegalStringEndQuote);
+
+	if (_isUnicode)
+	{
+		ScannerError unicodeDirectionError = UnicodeDirectionProcessor::parse(m_tokens[NextNext].literal);
```

Comment:
Doesn't the literal already have unicode escapes decoded?

Reply:
I think the safest way to make this work properly for everything is by using `CharStream::position()` to store the start position and then hand over `CharStream` to the processor which uses setPosition to re-process all the characters.